src/test/kotlin/se/uulm/snowballr/backend/repository/association/ProjectPaperTableRepoTest.kt

```diff
@@ -86,4 +115,57 @@ class ProjectPaperTableRepoTest : RepositoryTest(arrayOf(ProjectPaperTable, Proj
                 assertThat(projectPapers).noneMatch { it.projectPaper.id == nonProjectPaperId }
             }
     }
+
+    @Nested
+    inner class AddPaperToProject {
+        @Test
+        fun `When a project paper is added to a project, but the assigned user doesn't exist, then an exception is thrown`() =
+            runTest {
+                val request = Project.Paper.Add.newBuilder().build()
+                assertThrows<NotFoundException> { repo.addPaperToProject(request, UUID.randomUUID()) }
+            }
+
+        @Test
+        fun `When a project paper is added to a project with the correct values, then the project paper is returned`() =
+            runTest {
+                val paperId = insertPaperAndGetId()
+                val projectId = insertProjectAndGetId(createdBy = testUserId)
+                val request = Project.Paper.Add.newBuilder()
+                    .setPaperId(paperId.toString())
+                    .setProjectId(projectId.toString())
+                    .setStage(0)
+                    .build()
+
+                val projectPaper = assertDoesNotThrow { repo.addPaperToProject(request, testUserId) }
+                assertEquals(paperId, projectPaper.paperId)
+                assertEquals(projectId, projectPaper.projectId)
+                assertEquals(0, projectPaper.localPaperId)
+                assertEquals(0, projectPaper.stage)
+                assertEquals(ProjectOuterClass.PaperDecision.PAPER_DECISION_UNSPECIFIED, projectPaper.decision)
+                assertEquals(testUserId, projectPaper.createdBy)
+            }
+
+        @Test
+        fun `When a second project paper is added to a project with the correct values, then the local paper ID is increased correctly`() =
```

Comment:
```suggestion
        fun `When a second project paper is added to the same project with the correct values, then the local paper ID is increased correctly`() =
```

Also, let's add another test that adds a paper to a different project to see that the local IDs are project-local and don't depend on each other.

Reply:
Good idea.